docs/source/argument_check.rst

```diff
@@ -0,0 +1,607 @@
+=================================================
+rocBLAS order of argument checking and logging
+=================================================
+
+Legacy BLAS 
+===========
+Legacy BLAS has two types of argument checking:
+
+1. Error-return for incorrect argument (Legacy BLAS implement this with a call to the function ``XERBLA``).
+
+2. Quick-return-success when an argument allows for the subprogram to be a no-operation or a constant result.
+
+Level 1 BLAS subprograms have only quick-return-success. Level 2 and Level 3 BLAS subprograms have both quick-return-success and error-return. 
+
+rocBLAS
+=======
+rocBLAS has 4 types of argument checking:
+
+1. ``rocblas_status_invalid_handle`` if the handle is a NULL pointer
+
+2. ``rocblas_status_invalid_pointer`` for NULL argument pointers
+
+3. ``rocblas_status_invalid_size`` for invalid size, increment or leading dimension argument
```

Comment:
Now have invalid_value which should be described:  E.g.  if a rocblas enum which isn't supported is passed.

Reply:
Thanks,  corrected.